src/mod/applications/mod_av/avformat.c

```diff
@@ -2229,6 +2227,29 @@ GCC_DIAG_ON(deprecated-declarations)
 						char ebuf[255] = "";
 						switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "Error while writing audio frame: %s\n", get_error_text(ret, ebuf, sizeof(ebuf)));
 					}
+					
+					if(ret == -(ETIMEDOUT)) {
+						context->closed = 1;
+						status = SWITCH_STATUS_FALSE;
+						
+						if(!zstr(context->owner_uuid)) {
+							switch_core_session_t *session = switch_core_session_locate(context->owner_uuid);
+						
+							if(session) {
+								if(!zstr(handle->original_file_path) && (ret = switch_ivr_stop_record_session(session, handle->original_file_path)) != SWITCH_STATUS_SUCCESS) {
```

Comment:
I don't recommend calling `switch_ivr_stop_record_session` here. Take note that this is based on Freeswitch File Implementation, meaning I can also used this in playback. I would suggest to only return an error here, update the record_callback in switch_ivr_async and terminate when write fails. We should have a flipper as well to only do this if `RECORD_FAIL_ON_WRITE` is set to true. When we return error in media bug it should close the media bug by marking it as prune

Reply:
There is no need for a feature flipper.  If rw_timeout is not passed, then avio_open() is used, and the whole code path is moot.

I am not understanding the actual concern regarding the use of `switch_ivr_stop_record_session()` here.  Is there ever a situation where:

1.  We have an rtmp stream up that is being used for BOTH reading (playback) and writing (recording) AND
2.  Where if the socket timed out (note that this only currently happens on -(ETIMEOUT), meaning -110, for some reason they used the negative of the standard error value) for writing, that it somehow by magic the socket would still be up for reading (playback) ?  (This scenario is impossible, btw.)

Presumably (I didn't have a chance to check it yet), but upon calling `switch_ivr_stop_record_session()` , an event would be fired that call control could react upon in real time to re-issue a uuid_record command on that call.  (I've not received feedback from @bluesbettle  if that is the desired behavior or if it better to just set a chanvar that would indicate incomplete recording so post-processing could be done on the fullcap).